.github/workflows/emscripten.yml

```diff
@@ -52,14 +52,23 @@ jobs:
           actions-cache-folder: emsdk-cache
 
       - name: Install pyodide-build
-        run: pip install "pydantic<2" pyodide-build==$PYODIDE_VERSION
+        run: pip install "pydantic<2" build pyodide-build==$PYODIDE_VERSION
 
       - name: Build
         run: |
-          # Pyodide is still in the process of adding better/easier support for
-          # non-setup.py based builds.
-          cp pyproject.toml.setuppy pyproject.toml
-          CFLAGS=-g2 LDFLAGS=-g2 pyodide build
+          # Note that we need to cross-compile here. Using `pyodide build`
+          # fails on not having a cross file. The `source() function in
+          # `pyodide_build/cli/build.py` needs a --backend-flags parameter so
+          # that we can pass config-settings arguments to the backend.
+          # We'll then run into static assert issues in `pyconfig.h` due to:
+          # https://github.com/pyodide/pyodide/pull/2494#issuecomment-1118500743
+          # See also https://github.com/pyodide/pyodide/pull/2238 and the patches
+          # to _numpyconfig.h in https://github.com/pyodide/pyodide/tree/main/packages/numpy
+          #
+          # tl;dr this needs some work, and it seems like cross-compilation is
+          # hard because Pyodide doesn't ship its own Python interpreter that
+          # we can actually target during a direct cross build.
+          python -m build --wheel -Csetup-args=--cross-file=$PWD/tools/ci/emscripten.meson.cross -Csetup-args=-Dallow-noblas=true
```

Comment:
The `-Dallow-noblas=true` is no longer needed, the fallback to internal code when BLAS is missing is automatic.

Reply:
The `-Csetup-args=--cross-file=$PWD/tools/ci/emscripten.meson.cross` will still be needed I think, only to add the longdouble format, just in slightly different format. From [this comment](https://github.com/numpy/numpy/pull/24603#issuecomment-1713793418) I think it should be:
```
pyodide build setup-args="--cross-file=$PWD/tools/ci/emscripten.meson.cross"
```

Reply:
Pyodide bundles emscripten.meson.cross used in in-tree build, so probably you can try that one too.

```
PYODIDE_MESON_CROSS_FILE=`pyodide config get meson_cross_file`
pyodide build -C "setup-args=\"${PYODIDE_MESON_CROSS_FILE}\""
```